### PR TITLE
Remove lowering and decompositions of zero_, zero, zeros_like... in favour of their references

### PR DESCRIPTION
**Pull request by lezcano**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #92071

The generated triton code is identical.

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: torch/_refs/__init__.py [hunk 1]
```diff
@@ -4673,10 +4673,50 @@ def full_like(
...
+    dtype: Optional[torch.dtype] = None,
+    layout: Optional[torch.layout] = None,
+    device: Optional[torch.device] = None,
+    pin_memory: bool = False,
+    requires_grad: bool = False,
+    memory_format: torch.memory_format = torch.preserve_format,
+) -> TensorLikeType:
+    return torch.full_like(
+        a,
+        True if (dtype or a.dtype) == torch.bool else 0,
```
Comment (peterbell10) [inline torch/_refs/__init__.py:4712]:
```suggestion
        True if (dtype or a.dtype) == torch.bool else 1,
```

File: torch/_refs/__init__.py [hunk 2]
```diff
@@ -4673,10 +4673,50 @@ def full_like(
...
+    dtype: Optional[torch.dtype] = None,
+    layout: Optional[torch.layout] = None,
+    device: Optional[torch.device] = None,
+    pin_memory: bool = False,
+    requires_grad: bool = False,
+    memory_format: torch.memory_format = torch.preserve_format,
+) -> TensorLikeType:
+    return torch.full_like(
+        a,
+        False if (dtype or a.dtype) == torch.bool else 0,
```
Comment (peterbell10) [inline torch/_refs/__init__.py:4689]: Why doesn't `False` work here like it did with `_refs.full_like`?

File: torch/_refs/__init__.py (hunk 1 above)
Comment (lezcano) [reply torch/_refs/__init__.py:4712]: lol

File: torch/_refs/__init__.py (hunk 2 above)
Comment (lezcano) [reply torch/_refs/__init__.py:4689]: It does, but the `False` did show in the triton code and it was a bit odd. This way it should be a bit more type-safe.

Comment (peterbell10) [reply torch/_refs/__init__.py:4689]: That sound like an inductor bug then, I would expect it to convert the literal to the tensor dtype.

**Comment by lezcano** [comment]
> Also prims.fill should probably be just a ref in terms of full_like

Ah, right, I see that `full` is implemented in terms of `fill`. Fine.

File: torch/_refs/__init__.py (hunk 2 above)
Comment (peterbell10) [reply torch/_refs/__init__.py:4689]: Fix for this in #92093

Comment (lezcano) [reply torch/_refs/__init__.py:4689]: fantastic, thank you!

**Review by ngimel — APPROVED** [review]

**Pull request closed without merging**